### PR TITLE
Reset version number to 0.1.0

### PR DESCRIPTION
**Pull request by kennylevinsen**
No need to follow systemd version numbers, so set project version to
0.1.0 and the soversion to 1.

Closes: https://github.com/emersion/basu/issues/25

### CODE REVIEW

**Comment by emersion** [comment]
Note that the symbol versioning is still systemd's: https://github.com/emersion/basu/blob/master/src/libsystemd/libsystemd.sym

But that's probably for the best.

File: meson.build [hunk 1]
```diff
@@ -10,7 +10,7 @@ project('basu', 'c',
   meson_version: '>= 0.54',
 )
 
-libsystemd_version = '0.23.0'
+soversion = 1
```
Comment (emersion) [inline meson.build:13]: We could set this to 0 instead.

File: meson.build [hunk 2]
```diff
@@ -442,7 +442,7 @@ endforeach
 libsystemd_sym_path = '@0@/@1@'.format(meson.current_source_dir(), libsystemd_sym)
 libsystemd = shared_library(
         'basu',
-        version : libsystemd_version,
+        soversion : soversion,
```
Comment (emersion) [inline meson.build:445]: I wonder whether we should use `meson.project_version()` here. I guess if we do a 0.2.0 it won't bump the soversion? Then better to use an explicit soversion I guess.

**Comment by kennylevinsen** [comment]
> Note that the symbol versioning is still systemd's: https://github.com/emersion/basu/blob/master/src/libsystemd/libsystemd.sym
> 
> But that's probably for the best?

On the other hand, those numbers' only use case is binary compatibility in a use-case where a binary was built against libsystemd.

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**